command_prompt.bash

```diff
@@ -3,237 +3,153 @@
-                upstream=$SEAFLY_GIT_BEHIND ;;
-            *"	0") # ahead of upstream
-                upstream=$SEAFLY_GIT_AHEAD ;;
-            *)	    # diverged from upstream
-                upstream=$SEAFLY_GIT_DIVERGED ;;
-            esac
-        fi
-
-        local spacer
-        if [[ -n $dirty || -n $staged || -n $stash || -n $upstream ]]; then
-            spacer=" "
-        fi
-        export SEAFLY_GIT_DETAILS=" $SEAFLY_GIT_PREFIX$branch$spacer\[$SEAFLY_ALERT_COLOR\]$dirty\[$SEAFLY_NORMAL_COLOR\]$staged$upstream\[$SEAFLY_GIT_COLOR\]$stash$SEAFLY_GIT_SUFFIX"
+    if (( GIT_PS1_SHOWUPSTREAM )); then
+        (( VCS_STATUS_COMMITS_AHEAD ))  && upstream=$SEAFLY_GIT_AHEAD
+        (( VCS_STATUS_COMMITS_BEHIND )) && upstream=$SEAFLY_GIT_BEHIND
+        (( VCS_STATUS_COMMITS_AHEAD && VCS_STATUS_COMMITS_BEHIND )) && upstream=$SEAFLY_GIT_DIVERGED
     fi
-}
 
-# Collate Git details using the optimized
-# [gitstatus](https://github.com/romkatv/gitstatus) command.
-#
-_git_details_optimized() {
-    if gitstatus_query && [[ $VCS_STATUS_RESULT == ok-sync ]]; then
-        local branch=$VCS_STATUS_LOCAL_BRANCH
-        if [[ -z $branch ]]; then
-            branch="detached*$(git rev-parse --short HEAD 2>/dev/null)"
-        fi
-
-        local dirty
-        local staged
-        if [[ $GIT_PS1_SHOWDIRTYSTATE != 0 &&
-              $VCS_STATUS_HAS_UNSTAGED == 1 ]]; then
-            dirty=$SEAFLY_GIT_DIRTY
-        fi
-        if [[ $GIT_PS1_SHOWDIRTYSTATE != 0 &&
-              $VCS_STATUS_HAS_STAGED == 1 ]]; then
-            staged=$SEAFLY_GIT_STAGED
-        fi
-
-        local stash
-        if [[ $GIT_PS1_SHOWSTASHSTATE != 0 && $VCS_STATUS_STASHES -gt 0 ]]; then
-            stash=$SEAFLY_GIT_STASH
-        fi
-
-        local upstream
-        if [[ $GIT_PS1_SHOWUPSTREAM != 0 ]]; then
-            if [[ $VCS_STATUS_COMMITS_AHEAD -gt 0 &&
-                  $VCS_STATUS_COMMITS_BEHIND -gt 0 ]]; then
-                upstream=$SEAFLY_GIT_DIVERGED
-            elif [[ $VCS_STATUS_COMMITS_AHEAD -gt 0 ]]; then
-                upstream=$SEAFLY_GIT_AHEAD
-            elif [[ $VCS_STATUS_COMMITS_BEHIND -gt 0 ]]; then
-                upstream=$SEAFLY_GIT_BEHIND
-            fi
-        fi
-
-        local spacer
-        if [[ -n $dirty || -n $staged || -n $stash || -n $upstream ]]; then
-            spacer=" "
-        fi
-        export SEAFLY_GIT_DETAILS=" $SEAFLY_GIT_PREFIX$branch$spacer\[$SEAFLY_ALERT_COLOR\]$dirty\[$SEAFLY_NORMAL_COLOR\]$staged$upstream\[$SEAFLY_GIT_COLOR\]$stash$SEAFLY_GIT_SUFFIX"
-    fi
+    [[ -n $dirty || -n $staged || -n $stash || -n $upstream ]] && spacer=" "
+    _seafly_git_details=" $SEAFLY_GIT_PREFIX$ref$spacer\[$SEAFLY_ALERT_COLOR\]$dirty\[$SEAFLY_NORMAL_COLOR\]$staged$upstream\[$SEAFLY_GIT_COLOR\]$stash$SEAFLY_GIT_SUFFIX"
 }
 
-_command_prompt() {
-    # Run a pre-command if set.
-    if [[ -n $SEAFLY_PRE_COMMAND ]]; then
-        eval $SEAFLY_PRE_COMMAND
-    fi
+_seafly_prompt_command() {
```

Comment:
Prefix all internal functions with `_seafly` to avoid clashes with other projects.